### PR TITLE
Add subcommand APP for mesheryctl

### LINKED ISSUE #3510: [mesheryctl] Add `mesheryctl app` sub-command to manage Meshery Applications
_See [mesheryctl Command Tracker](https://bit.ly/3dqXy1q) for current status of commands._

#### Current Behavior

Meshery has a new construct called "applications" which lets users define their applications/workloads in pattern files and use Meshery to store, and onboard these applications.

#### Desired Behavior

`mesheryctl` should also support these operations through a new `app` sub-command.

The following sub-commands should be added:

`app onboard` similar to `pattern apply`
`app offboard` similar to `pattern delete`
`app view` similar to `pattern view`
`app list` similar to `pattern list`

The current implementation of `pattern` sub-command can be copied and used here making the necessary changes. 

---

#### Contributor Resources

- [mesheryctl Contributing Guide](https://github.com/meshery/meshery/blob/master/mesheryctl/README.md)
- [Beginner's guide to contributing to Meshery and mesheryctl](https://youtu.be/hh_kFLZx3G4)
- [mesheryctl Command Tracker](https://docs.google.com/spreadsheets/d/1q63sIGAuCnIeDs8PeM-0BAkNj8BBgPUXhLbe1Y-318o/edit#gid=0)
- [Meshery CLI Commands and Documentation](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#heading=h.5fucij4hc5wt)
- [Layer5 Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by suhaniii**
**Description**

This PR fixes #3510 

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
Please sign-off on your commit(s). To do so for the commits in this PR, execute - https://github.com/meshery/meshery/pull/3520/checks?check_run_id=3078282571

To configure your system to always sign-off, see the instructions in the contributing.md readme.

**Comment by suhaniii** [comment]
> Please sign-off on your commit(s). To do so for the commits in this PR, execute - https://github.com/meshery/meshery/pull/3520/checks?check_run_id=3078282571
> 
> To configure your system to always sign-off, see the instructions in the contributing.md readme.

Not able to sign it with following 

```
git commit --amend --no-edit --signoff
git push --force-with-lease origin feature/suhaniii/3510
```

**Comment by Anita-ihuman** [comment]
@suhaniii, kindly reply in slack

**Comment by suhaniii** [comment]
> @suhaniii, kindly reply in slack

Hi @Anita-ihuman , Thanks for the checking up :) , I was able to pass DCO this morning.

**Comment by suhaniii** [comment]
@leecalcote @Anita-ihuman  @navendu-pottekkat This PR needs reviewers. Adding reviewers is disabled for me.

**Comment by piyushsingariya** [comment]
@leecalcote I think we already have this subcommand under `exp` by the name `application` in mesheryctl.

File: mesheryctl/internal/cli/root/app/list.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+		return nil
+
+	},
+}
+
+func init() {
+	listCmd.Flags().BoolVarP(&verbose, "verbose", "v", false, "Display full length user and app file identifiers")
+	listCmd.Flags().StringVarP(&token, "token", "t", "", "path to token")
+	_ = listCmd.MarkFlagRequired("token")
```
Comment (piyushsingariya) [inline mesheryctl/internal/cli/root/app/list.go:131]: Hey @suhaniii can you look into some latest changes in `pattern` command which help us using token without making it compulsory to pass `--token` flag.

**Review by piyushsingariya — CHANGES_REQUESTED** [review]

File: mesheryctl/internal/cli/root/app/list.go (hunk 1 above)
Comment (suhaniii) [reply mesheryctl/internal/cli/root/app/list.go:131]: Thanks. I will update it.

**Pull request closed without merging**